server/core/src/main/java/dev/slimevr/osc/VMCHandler.kt

```diff
@@ -70,108 +72,122 @@ class VMCHandler(
 	override fun refreshSettings(refreshRouterSettings: Boolean) {
 		anchorHip = config.anchorHip
 
-		// Stops listening and closes OSC port
+		updateOscReceiver(config.portIn, listenAddresses)
+		updateOscSender(config.portOut, config.address)
+
+		if (config.enabled) {
+			// Load VRM data
+			if (outputUnityArmature != null && config.vrmJson != null) {
+				val vrmReader = VRMReader(config.vrmJson!!)
+				for (unityBone in UnityBone.entries) {
+					val node = outputUnityArmature!!.getHeadNodeOfBone(unityBone)
+					if (node != null) {
+						node
+							.localTransform
+							.translation = vrmReader.getOffsetForBone(unityBone!!)
+					}
+				}
+				vrmHeight = vrmReader
+					.getOffsetForBone(UnityBone.HIPS)
+					.plus(vrmReader.getOffsetForBone(UnityBone.SPINE))
+					.plus(vrmReader.getOffsetForBone(UnityBone.CHEST))
+					.plus(vrmReader.getOffsetForBone(UnityBone.UPPER_CHEST))
+					.plus(vrmReader.getOffsetForBone(UnityBone.NECK))
+					.plus(vrmReader.getOffsetForBone(UnityBone.HEAD))
+					.len()
+			}
+		}
+
+		if (refreshRouterSettings) server.oSCRouter.refreshSettings(false)
+	}
+
+	override fun updateOscReceiver(portIn: Int, args: Array<String>) {
+		// Stop listening
 		val wasListening = oscReceiver != null && oscReceiver!!.isListening
 		if (wasListening) {
 			oscReceiver!!.stopListening()
 		}
-		val wasConnected = oscSender != null && oscSender!!.isConnected
-		if (wasConnected) {
-			try {
-				oscSender!!.close()
-			} catch (e: IOException) {
-				LogManager.severe("[VMCHandler] Error closing the OSC sender: $e")
-			}
-		}
 
 		if (config.enabled) {
 			// Instantiates the OSC receiver
 			try {
-				val port = config.portIn
-				oscReceiver = OSCPortIn(port)
-				if (lastPortIn != port || !wasListening) {
-					LogManager.info("[VMCHandler] Listening to port $port")
+				oscReceiver = OSCPortIn(portIn)
+				if (lastPortIn != portIn || !wasListening) {
+					LogManager.info("[VMCHandler] Listening to port $portIn")
 				}
-				lastPortIn = port
+				lastPortIn = portIn
 			} catch (e: IOException) {
 				LogManager
 					.severe(
-						"[VMCHandler] Error listening to the port ${config.portIn}: $e",
+						"[VMCHandler] Error listening to the port " +
+							portIn +
+							": " +
+							e,
```

Comment:
why, why return to monke
```suggestion
						"[VMCHandler] Error listening to the port $portIn: $e",
```